assets/js/modules/analytics-4/datastore/custom-dimensions.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * `modules/analytics-4` data store: custom-dimensions store.
+ *
+ * Site Kit by Google, Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+import { isPlainObject } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import Data from 'googlesitekit-data';
+import { createFetchStore } from '../../../googlesitekit/data/create-fetch-store';
+import { createReducer } from '../../../googlesitekit/data/create-reducer';
+import { isValidPropertyID } from '../utils/validation';
+
+const customDimensionFields = [
+	'parameterName',
+	'displayName',
+	'description',
+	'scope',
+	'disallowAdsPersonalization',
+];
+
+const fetchCreateCustomDimensionStore = createFetchStore( {
+	baseName: 'createCustomDimension',
+	controlCallback: ( { propertyID, customDimension } ) =>
+		API.set( 'modules', 'analytics-4', 'create-custom-dimension', {
+			propertyID,
+			customDimension: {
+				...customDimension,
+				scope: 'EVENT',
+			},
+		} ),
+	reducerCallback: createReducer( ( state, customDimension ) => {
+		state.customDimensions.push( customDimension );
```

Comment:
This seems reasonable but I'm not sure about this state since we're going to keep track of available custom dimensions in a module setting. I can see how this could be useful, I'm just not aware of a plan for its use. @nfmohit ?

We could potentially add the parameter name to the list of available dimensions in the setting to unblock things that will depend on that state sooner, otherwise those would be blocked until we sync the available dimensions (which is where we'd refresh the persisted list).

Reply:
What @aaemnnosttv said. @hussain-t and I just also had a discussion about this internally. I think the `customDimensions` state here will not add any value, so this can be removed.

Reply:
Thanks @aaemnnosttv and @nfmohit. I have removed the state from the store.